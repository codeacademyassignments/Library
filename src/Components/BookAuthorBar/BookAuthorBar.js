import React, { Component } from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import style from './BookAuthorBar.style';

export default class BookAuthorBar extends Component {
  getAuthorComponents = (authorArray) => {
    const authorTextArray = authorArray.slice(0, 2).reduce((accumulator, current) => {
      accumulator.push(<Text key={current} style={style.author}>{current}</Text>);
      return accumulator;
    }, []);
    return authorTextArray;// limiting to two
  }

  render() {
    const { authorArray } = this.props;
    return (
      <View style={style.authorNames}>
        {this.getAuthorComponents(authorArray)}
      </View>
    );
  }
}

BookAuthorBar.propTypes = {
  authorArray: PropTypes.array.isRequired,
};
