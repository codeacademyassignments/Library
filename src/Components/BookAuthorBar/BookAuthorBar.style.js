import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  authorNames: {
    fontWeight: '100',
    flex: 1,
    flexDirection: 'row',
  },
  author: {
    fontSize: 10,
    marginRight: 10,
  },
});
