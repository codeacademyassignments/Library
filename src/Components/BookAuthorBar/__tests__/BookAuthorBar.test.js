import renderer from 'react-test-renderer';
import React from 'react';
import { Text } from 'react-native';
import BookAuthorBar from '../BookAuthorBar';
import style from '../BookAuthorBar.style';

const authorNameArray = ['J K Rowling', 'Paulo Coelho', 'Will Smith'];

describe('BookAuthorBar', () => {
  it('should match old snapshot', async () => {
    const tree = renderer.create(<BookAuthorBar authorArray={authorNameArray} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('getAuthorComponents', () => {
  it('should return array with two rows each having Text component', async () => {
    const tree = renderer.create(<BookAuthorBar authorArray={authorNameArray} />);
    expect(tree.getInstance().getAuthorComponents(authorNameArray).length).toEqual(2);
  });
  it('should return empty array when passed author array os empty', async () => {
    const tree = renderer.create(<BookAuthorBar authorArray={authorNameArray} />);
    expect(tree.getInstance().getAuthorComponents([])).toEqual([
    ]);
  });
});
