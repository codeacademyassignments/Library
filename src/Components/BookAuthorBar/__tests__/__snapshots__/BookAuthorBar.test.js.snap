// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookAuthorBar should match old snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "fontWeight": "100",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 10,
        "marginRight": 10,
      }
    }
  >
    J K Rowling
  </Text>
  <Text
    style={
      Object {
        "fontSize": 10,
        "marginRight": 10,
      }
    }
  >
    Paulo Coelho
  </Text>
</View>
`;
